arches/app/models/tile.py

```diff
@@ -537,11 +537,14 @@ def delete(self, *args, **kwargs):
                         node = SimpleNamespace(**next((x for x in self.serialized_graph["nodes"] if x["nodeid"] == nodeid), None))
                     except TypeError: #will catch if serialized_graph is None
                         node = models.Node.objects.get(nodeid=nodeid)
+
                     datatype = self.datatype_factory.get_instance(node.datatype)
                     datatype.post_tile_delete(self, nodeid, index=index)
-                if index:
```

Comment:
This may be necessary, but I'm concerned about creating a resource proxy model instance for every tile save whether or not we need to index. That would decrease the performance of loading data using the cli and probably in other cases as well.

Reply:
This is on the tile.delete method

Reply:
Thanks @apeters updated my comment - I meant that it applies to both save and delete.

Reply:
@chiatt even though `save_descriptors` gets called on every save or delete it won't mean that the descriptors will be recalculated every time.  I think they only time they get recalculated is when you save or delete a tile with `sortorder = 0`

Reply:
not in the code's current state @apeters. We need to handle the case where a cardinality `n` nodegroup  has multiple tiles and the user deletes the first tile.

Reply:
@chrabyrd I just checked and confirmed that delete of the first tile updates the primary name with the second tile. I thought that was the case because of lines 45-48 in the primary_descriptors function.

Reply:
@apeters  Yes, this is because we are recalculating the descriptors on every tile save, regardless of `sortorder`